File: Py/08_exception_handling.py
# Exception Handling
x, y = 5, 0

# Use assert to check if conditions are met
assert y != 0, "y cannot be zero"

# Use simple if statement to check conditions and raise errors
if y == 0:
    raise ZeroDivisionError("y is not allowed to be 0")

# Simple try catch
try:
    z = x / y

except ZeroDivisionError as e:
    print("Divide by zero error")
    print(e)

else:
    print("Else executed")

finally:
    print("Final line executed")

File: Cpp/08_exception_handling.cpp
#include <iostream> 
#include <exception>
#include <assert.h>
  
int main() 
{ 
    int x = 5; 
    int y = 0;
    int z;

    assert(y != 0);
    
    try { 
        if (y == 0){ 
            throw "Divide by zero error";
        } 
        z = x / y;
    } catch (int x) { 
        std::cout << "Exception Caught x: " << x << std::endl; 
    } catch (const char* error) {
        std::cout << error << std::endl;
    } catch (std::exception& e) {
        std::cout << "Standard exception: " << e.what() << std::endl;
    } catch(...) {
        std::cout << "Any exception caught!" << std::endl;
    }

    std::cout << "Final line executed" << std::endl; 
    return 0; 
} 
